### PR TITLE
61455 oauth as a service

### PR DESCRIPTION
**Pull request by nilsgstrabo**

### CODE REVIEW

**Review by satr — CHANGES_REQUESTED** [review]

File: pkg/apis/deployment/deployment.go [hunk 1]
```diff
@@ -539,5 +564,12 @@ func (deploy *Deployment) syncDeploymentForRadixComponent(component v1.RadixComm
 			return fmt.Errorf("failed to delete servicemonitor: %v", err)
 		}
 	}
+
+	// err = deploy.oauthProxyResourceManager.Sync(component)
+	// if err != nil {
+	// 	log.Infof("Failed to sync oauth2 proxy: %v", err)
+	// 	return fmt.Errorf("failed to sync oauth2 proxy: %v", err)
+	// }
+
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply pkg/apis/deployment/deployment.go]: fixed

File: pkg/apis/deployment/environmentvariables.go [hunk 2]
```diff
@@ -97,6 +97,7 @@ func getEnvironmentVariablesFrom(kubeutil *kube.Kube, appName string, envVarsSou
 		return nil, err
 	}
 
+	// return getEnvironmentVariables(appName, envVarsSource, radixDeployment, deployComponent.GetName(), deployComponent.GetSecrets(), deployComponent.IsPublic(), deployComponent.GetPorts(), envVarsConfigMap, deployComponent.GetEnvironmentVariables()), nil
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply pkg/apis/deployment/environmentvariables.go]: fixed

File: pkg/apis/deployment/ingressannotationprovider.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+func NewIngressConfigurationAnnotationProvider(config IngressConfiguration) IngressAnnotationProvider {
+	return &ingressConfigurationAnnotationProvider{config: config}
+}
+
+type ingressConfigurationAnnotationProvider struct {
+	config IngressConfiguration
+}
+
+func (o *ingressConfigurationAnnotationProvider) GetAnnotations(component v1.RadixCommonDeployComponent) (map[string]string, error) {
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply pkg/apis/deployment/ingressannotationprovider.go]: fixed

File: pkg/apis/deployment/radixcomponent.go [hunk 4]
```diff
@@ -64,45 +73,33 @@ func getRadixComponentAlwaysPullImageOnDeployFlag(radixComponent *v1.RadixCompon
...
 	} else if componentAuthentication == nil {
-		authentication = environmentAuthentication.DeepCopy()
+		return environmentAuthentication.DeepCopy(), nil
 	} else if environmentAuthentication == nil {
-		authentication = componentAuthentication.DeepCopy()
-	} else {
-		authentication = &v1.Authentication{
-			ClientCertificate: GetClientCertificateForComponent(componentAuthentication.ClientCertificate, environmentAuthentication.ClientCertificate),
-		}
+		return componentAuthentication.DeepCopy(), nil
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply pkg/apis/deployment/radixcomponent.go]: Removed all else-if to make code more readable (hopefully)

File: pkg/apis/utils/oauth/oauth.go [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package oauth
+
+import (
+	"net/url"
+	"path"
+	"strings"
+)
+
+// SanitizePathPrefix adds a leading "/" and strips away any "/" at the end
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply pkg/apis/utils/oauth/oauth.go]: fixed

File: pkg/apis/defaults/oauth2.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
+package defaults
+
+import (
+	commonUtils "github.com/equinor/radix-common/utils"
+	mergoutils "github.com/equinor/radix-common/utils/mergo"
+	v1 "github.com/equinor/radix-operator/pkg/apis/radix/v1"
+	"github.com/imdario/mergo"
+)
+
+type OAuth2Config interface {
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply pkg/apis/defaults/oauth2.go:13]: fixed

File: pkg/apis/utils/oauth/oauth.go [hunk 7]
```diff
@@ -0,0 +1,22 @@
+package oauth
+
+import (
+	"net/url"
+	"path"
+	"strings"
+)
+
+// SanitizePathPrefix adds a trailing "/" and strips away any "/" at the end
```
Comment (satr) [inline pkg/apis/utils/oauth/oauth.go:9]: I would expected something like:
'// SanitizePathPrefix adds a leading "/" and strips away any trailing "/" ' :)

**Review by satr — CHANGES_REQUESTED** [review]

File: pkg/apis/utils/oauth/oauth.go (hunk 7 above)
Comment (nilsgstrabo) [reply pkg/apis/utils/oauth/oauth.go]: fixed

File: pkg/apis/utils/random.go [hunk 8]
```diff
@@ -22,6 +24,15 @@ func hash(s string) int64 {
 	return int64(nr)
 }
 
+// GenerateRandomKey generates a cryptographically secure random key
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply pkg/apis/utils/random.go]: moved

**Review by satr — APPROVED** [review]

**Pull request merged**